js/ElementRendererProvider.js

```diff
@@ -528,6 +550,98 @@ export default class ElementRendererProvider {
+        Utils.appendAttributesFromObject(carouselWrapper, config.accessibility.web);
+      }
+
+      function findCardParent(element: any): HTMLDivElement | typeof undefined {
+        if (!element || element.tagName === 'BUTTON') {
+          return undefined;
+        }
+
+        const index = element.getAttribute('data-carousel-index');
+        if (index !== null) {
+          return element;
+        }
+
+        return findCardParent(element.parentNode);
+      }
+
+      function toggleCardSelect(element: HTMLDivElement, selected: boolean) {
+        if (selected) {
+          element.setAttribute('data-selected', 'true');
+          element.classList.add('lp-json-pollock-layout-selected');
+        } else {
+          element.removeAttribute('data-selected');
+          element.classList.remove('lp-json-pollock-layout-selected');
+        }
+      }
+
+      /**
+       * @param {MouseEvent} event
+       * */
+      function cardClick(event: MouseEvent) {
+        const element = event.target;
+        const cardParent = findCardParent(element);
+
+        if (cardParent) {
+          if (config.selectMode.type === 'single' && cardParent.parentNode) {
+            Array
+              .from((cardParent.parentNode: any).querySelectorAll('[data-carousel-index][data-selected]'))
+              .filter((carouselElement: HTMLDivElement) => carouselElement !== cardParent)
+              .forEach((carouselElement: HTMLDivElement) => {
+                toggleCardSelect(carouselElement, false);
+              });
+          }
+
+          toggleCardSelect(cardParent, cardParent.dataset.selected !== 'true');
+        }
+      }
+
+      /**
+       * Render logic
+       * */
+      (carouselWrapper: any).afterRender = () => {
+        const carouselItemsCount = carouselWrapper.childNodes.length;
+
+        if (carouselItemsCount) {
+          for (let itemCounter = 0;
+               itemCounter < carouselItemsCount;
+               itemCounter += 1) {
+            const carouselElement: HTMLDivElement =
+              (carouselWrapper.childNodes[itemCounter]: any);
+
+            carouselElement.addEventListener('click', cardClick.bind(this), true);
+            carouselElement.style.margin = `0 ${padding / 2}px`; // this comment is due to a bug in VSCode js editor :( otherwise ut shows the code below as a comment `
+            carouselElement.setAttribute('data-carousel-index', itemCounter.toString());   // Add an index reference for faster lookup on focus changes
+            carouselElement.setAttribute('role', 'listitem');
+          }
+
+          /* create carousel wrapper */
+          while (carouselWrapper.hasChildNodes() && carouselWrapper.lastChild) {
+            carousel.insertBefore(carouselWrapper.lastChild, carousel.firstChild);
+          }
+
+          carousel.className = 'lp-json-pollock-layout-carousel lp-json-pollock-layout-carousel-select';
+          carousel.setAttribute('aria-label', 'Carousel with buttons');
```

Comment:
Can we use [accessibilityWeb](https://gitlab.com/l1905/conversational-cloud-engineering/conversation-exchange-services/unified-messaging-system/json-schema/-/blob/master/json-schema-files/src/main/resources/json_schemas/rich/accessibilityWeb.json) for this?

Reply:
Yes, sure! Thank you for the comment.
But we have it at the 560-562 lines. So I must remove these extra `aria-label` and `role` attributes, right?

Reply:
Yeah, basically set this value using the accessiblityWeb object instead of hard-coding, if possible

Reply:
It was removed, and done. Thank you!